### PR TITLE
[WFLY-17268] Drop source transformation for the weld/subsytem module

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.redhat.com/browse/WFLY-17268

### CODE REVIEW

File: weld/subsystem/src/main/java/org/jboss/as/weld/deployment/AlternativePropertyReplacingBeansXmlParser.java
Comment (bstansberry) [inline weld/subsystem/src/main/java/org/jboss/as/weld/deployment/AlternativePropertyReplacingBeansXmlParser.java:34]: Note that I intend to deal with this in a follow up commit, but I don't want the basic work to block on it.

If I get a chance to do that before this PR is approved, then I'll add that to this PR, otherwise I'll use a separate PR.

It's not a big deal to do; it's just been a busy week so haven't gotten to it.

**Comment by manovotn** [comment]
@bstansberry IIUIC, this means that you can now fully depend on Weld 5/CDI 4, along with build compatible extensions.
In that case, you might also want to take a look at these places:

* https://github.com/wildfly/wildfly/blob/main/weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/WeldPortableExtensionProcessor.java#L61-L63
* https://github.com/wildfly/wildfly/blob/main/weld/subsystem/src/main/java/org/jboss/as/weld/discovery/WeldClassFileInfo.java#L130
* https://github.com/wildfly/wildfly/blob/main/weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/BeanArchiveProcessor.java#L89

And on a slightly different note, this piece can also be removed as there is a release that contains the change:
* https://github.com/wildfly/wildfly/blob/main/weld/subsystem/src/main/java/org/jboss/as/weld/deployment/WeldDeployment.java#L305-L313

All of the above should be trivial changes that I think you could include here as well.

File: weld/subsystem/src/main/java/org/jboss/as/weld/deployment/AlternativePropertyReplacingBeansXmlParser.java
Comment (manovotn) [reply weld/subsystem/src/main/java/org/jboss/as/weld/deployment/AlternativePropertyReplacingBeansXmlParser.java:34]: Whichever way suits you, but I wouldn't mind it being part of this PR along with the changes I mentioned in my comment so that we get that out of the way and don't forget it.

**Comment by bstansberry** [comment]
Thanks @manovotn -- I had time today so I made all those changes.

**Review by manovotn — APPROVED** [review]
Looking good, thanks @bstansberry!

**Pull request merged**
